pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -0,0 +1,113 @@
+import warnings
+from functools import singledispatch
+
+# import jax
+# import jax.numpy as jnp
+import torch
+import numpy as np
+
+from pytensor.compile.ops import DeepCopyOp, ViewOp
+from pytensor.configdefaults import config
+from pytensor.graph.fg import FunctionGraph
+from pytensor.ifelse import IfElse
+from pytensor.link.utils import fgraph_to_python
+from pytensor.raise_op import Assert, CheckAndRaise
+
+
+# if config.floatX == "float64":
+#     jax.config.update("jax_enable_x64", True)
+# else:
+#     jax.config.update("jax_enable_x64", False)
+
+
+@singledispatch
+def pytorch_typify(data, dtype=None, **kwargs):
+    r"""Convert instances of PyTensor `Type`\s to PyTorch types."""
+    if dtype is None:
+        return data
+    else:
+        return torch.tensor(data, dtype=dtype)
+
+
+@pytorch_typify.register(np.ndarray)
+def pytorch_typify_ndarray(data, dtype=None, **kwargs):
+    if len(data.shape) == 0:
+        return data.item()
```

Comment:
This looks strange. Why would we want to return a scalar instead of a rank 0 array?
The jax typify does the same thing, so maybe there is a good reason, but I'd like to know at least :-)

Reply:
JAX does it because of the many functions that only accept python integers as arguments, I'm not sure it should have been done here or at the dispatch level. 

We can try without the hack first for pytorch